packages/lib/models/Note.test.ts

```diff
@@ -332,4 +332,57 @@ describe('models_Note', function() {
 		expect(sortedNotes3[4].id).toBe(note2.id);
 	}));
 
+	it('should copy conflicted note to target folder and cancel conflict', (async () => {
+		// Prepare
+		const srcfolder = await Folder.save({ title: 'Source Folder' });
+		const targetfolder = await Folder.save({ title: 'Target Folder' });
+
+		const note1 = await Note.save({ title: 'note', parent_id: srcfolder.id });
+
+		const tmpConflictedNote = Object.assign({}, note1);
+		delete tmpConflictedNote.id;
+		tmpConflictedNote.is_conflict = 1;
+		tmpConflictedNote.conflict_original_id = note1.id;
+		const conflictedNote = await Note.save(tmpConflictedNote, { autoTimestamp: false });
```

Comment:
We don't copy and paste large blocks of code in this codebase, it's just a mess to maintain otherwise. So please do this:

- Create a function in models/Note called `async createConflictNote(sourceNote:NoteEntity, changeSource:number):NoteEntity`
- Make it create the conflict note and save it as in the code above.
- Then use that function in:
    - Your two test units
    - In lib/Synchronizer.ts

As a general rule if you see you are copying more than two or three lines of code, you should stop and think if the code can be refactored. Such refactoring improves the code quality because it wraps and names (using the function name) business logic and makes it re-usable.

Reply:
Should be good now.